int intensity
int x
int y
---
string cmdstr